File: Stack/java/Stack.java
public class Stack<T> {
	private int capacity = 50;

	private int index = -1;

	private T[] stack;

	public Stack() {
		stack = (T[])new Object[capacity];
	}

	public void push(T data) {
		stack[++index] = data;
	}

	public T pop() {
		return stack[index--];
	}

	public boolean isEmpty() {
		if (index == 0) {
			return true;
		} else {
			return false;
		}
	}

	public int size() {
		return index + 1;
	}

	public T peek() {
		return stack[index];
	}
}

File: Stack/kotlin/Stack.kt
//Unlike the Java Stack, this is a stack for Ints only, it isn't generic. The code for that was a little too inovlved for me at the moment
class Stack() {

	val capacity : Int = 50

	var index : Int = -1

	val stack = IntArray(capacity)

	fun push(input : Int) {
		stack[++index] = input;
	}

	fun pop() : Int{
		return stack[index--]
	}

	fun peek() : Int {
		return stack[index]
	}

	fun isEmpty() : Boolean {
		if (index == 0) {
			return true
		} else {
			return false
		}
	}

	fun size() : Int {
		return index + 1
	}
}
